src/hooks/useCardForm.tsx

```diff
@@ -0,0 +1,127 @@
+import useCardNumber from './useCardNumber';
+import useCardValidityPeriod from './useCardValidityPeriod';
+import useCardCVC from './useCardCVC';
+import useCardCompany from './useCardCompany';
+import useCardPassword from './useCardPassword';
+import useStep from './useStep';
+
+const useCardForm = () => {
+  const {
+    cardNumber,
+    onChangeCardNumber,
+    errorMessage: cardNumberErrorMessage,
+    setInputRef: setCardNumberInputRef,
+  } = useCardNumber();
+
+  const {
+    cardValidityPeriod,
+    onChangeMonth,
+    onChangeYear,
+    errorMessage: cardValidityPeriodErrorMessage,
+    setInputRef: setCardValidityPeriodInputRef,
+  } = useCardValidityPeriod();
+
+  const {
+    cardCVC,
+    onChangeCVC,
+    errorMessage: cardCVCErrorMessage,
+  } = useCardCVC();
+
+  const {
+    selectedCard,
+    CARD_COMPANY_NAMES,
+    CARD_COMPANY_COLORS,
+    CARD_COMPANY_PLACEHOLDER,
+    onClickCardCompany,
+  } = useCardCompany();
+
+  const {
+    cardPassword,
+    onChangeCardPassword,
+    errorMessage: cardPasswordErrorMessage,
+  } = useCardPassword();
+
+  const getErrorMessageFromList = (errorMessageList: string[]) => {
+    const filteredErrorMessageList = errorMessageList.filter(
+      (errorMessage) => errorMessage !== '',
+    );
+
+    if (filteredErrorMessageList.length === 0) {
+      return '';
+    }
+    return filteredErrorMessageList[0];
+  };
+
+  const formStatus = {
+    cardNumberOkay:
+      cardNumber.every((value) => value !== '') &&
+      !getErrorMessageFromList(cardNumberErrorMessage),
+    cardCompanyOkay: selectedCard !== null,
+    cardValidityPeriodOkay:
+      Object.values(cardValidityPeriod).every((value) => value !== '') &&
+      !getErrorMessageFromList(Object.values(cardValidityPeriodErrorMessage)),
+    cardCVCOkay: cardCVC !== '' && !cardCVCErrorMessage,
+    cardPasswordOkay: cardPassword !== '' && !cardPasswordErrorMessage,
+  };
+
+  const {
+    cardNumberOkay,
+    cardCompanyOkay,
+    cardValidityPeriodOkay,
+    cardCVCOkay,
+  } = formStatus;
+
+  const canSubmit = Object.values(formStatus).every(Boolean);
```

Comment:
key 값과 식별자에 대한 차이는 있겠지만, 네이밍 규칙이 통일되면 더 파악하기 쉽지 않을까하는 생각이 있어요.

Reply:
[각 필드가 유효한지 검증 하는 로직을 각각의 훅으로 분리](https://github.com/woowacourse/react-payments/pull/494#discussion_r2071942043)했더니 useCardForm에서는 정말 값만 사용하면 돼서 canSubmit도 더 깔끔하게 만들 수 있었습니다!

```suggestion
  const canSubmit =
    isCardNumberValid &&
    isCardCompanyValid &&
    isCardValidityPeriodValid &&
    isCardCVCValid &&
    isCardPasswordValid;
```

Reply:
cardNumberOkay
isError
등 boolean에 대한 네이밍이 상이한 점을 말씀드렸어요!